lib/pages/add_bridge/add_bridge_body.dart

```diff
@@ -27,40 +28,66 @@ class AddBridgeBody extends StatefulWidget {
 
 class _AddBridgeBodyState extends State<AddBridgeBody> {
   late BotBridgeConnection botConnection;
-
+  late String hostname;
   bool timeoutErrorOccurred = false;
 
   @override
   void initState() {
     final client = Matrix.of(context).client;
-    botConnection = BotBridgeConnection(client: client);
+    String fullUrl = client.homeserver!.host;
+    hostname = extractHostName(fullUrl);
+    botConnection = BotBridgeConnection(client: client, hostname: hostname);
     super.initState();
     _initStateAsync();
   }
 
-// Online status update when page is opened
+  @override
+  void dispose() {
+    botConnection.stopProcess();
+    super.dispose();
+  }
+
+  // Online status update when page is opened
   Future<void> _initStateAsync() async {
     try {
       final instagramConnected = await botConnection.pingWithTimeout(
-          context, botConnection.instagramPing());
-      setState(() {
-        socialNetwork
-            .firstWhere((element) => element.name == "Instagram")
-            .connected = instagramConnected;
-        socialNetwork
-            .firstWhere((element) => element.name == "Instagram")
-            .loading = false;
-      });
+        context,
+        botConnection.instagramPing(),
+      );
+      if (!mounted) return; // Check if the widget is still mounted
+
+      if (instagramConnected != 'error') {
+        setState(() {
+          socialNetwork
+              .firstWhere((element) => element.name == "Instagram")
+              .connected = instagramConnected == 'Connected' ? true : false;
+          socialNetwork
+              .firstWhere((element) => element.name == "Instagram")
+              .loading = false;
+        });
+      } else {
```

Comment:
```suggestion
      } else if (mounted) {
```
and minor changes below